### PR TITLE
Alerting: Fix bug when saving a rule more than once

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR resolves a bug in the alert rule form that occurred when saving multiple times without leaving the form.

**Why do we need this feature?**

It's a bug.
**Who is this feature for?**

All users.

**Special notes for your reviewer:**

Before the fix:

https://github.com/user-attachments/assets/c29236a3-e6f0-4835-9f7a-b8d03ef0c7dd

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/components/rule-editor/alert-rule-form/AlertRuleForm.tsx [hunk 1]
```diff
@@ -184,13 +184,15 @@ export const AlertRuleForm = ({ existing, prefill }: Props) => {
 
       locationService.push(returnToUrl);
       return;
-    }
+    } else {
+      // we stay in the same page
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/rule-editor/alert-rule-form/AlertRuleForm.tsx:188]: this is from the previous commit, but i believe it's ok

**Review by konrad147 — APPROVED** [review]
LGTM!

**Pull request merged**